src/devices/src/virtio/block/request.rs

```diff
@@ -18,28 +18,61 @@ use super::device::{CacheType, DiskProperties};
 use super::{Error, SECTOR_SHIFT, SECTOR_SIZE};
 
 #[derive(Debug)]
-pub enum ExecuteError {
+pub enum IoErrStatus {
     BadRequest(Error),
     Flush(io::Error),
-    Read(GuestMemoryError),
+    Read(u32, GuestMemoryError),
     Seek(io::Error),
     SyncAll(io::Error),
     Write(GuestMemoryError),
+}
+
+#[derive(Debug)]
+pub enum ErrStatus {
```

Comment:
nit: I would rename this to something like `StatusError`. Actually all `Err*` stuff.